app/controllers/user_funded_projects_controller.rb

```diff
@@ -5,4 +5,20 @@ def new
     @project = Project.find(params[:project_id])
   end
 
+  def create
+    @user = current_user
+    @project = Project.find(params[:project_id])
+    fund = @user.user_funded_projects.create(project_params)
+    @role = Role.find_or_create_by(name: "project_funder")
+    @user.roles << @role
+    @project.funders << @user
```

Comment:
Although I'm not sure if this line is necessary as we establish that relationship on line 11.